src/js/components/CheckBox/CheckBox.js

```diff
@@ -30,50 +34,105 @@ class CheckBox extends Component {
       ...rest
     } = this.props;
 
-    const normalizedLabel = (typeof label === 'string' ? <div>{label}</div> : label);
-
     let hidden;
     if (disabled && checked) {
       hidden = <input name={name} type='hidden' value='true' />;
     }
 
     const Icon = theme.checkBox.icons.checked;
 
+    let borderColor = normalizeColor(
+      theme.checkBox.border.color[theme.dark ? 'dark' : 'light'],
+      theme
+    );
+    if (checked) {
+      borderColor = normalizeColor(
+        (
+          theme.checkBox.color || theme.global.control.color
+        )[theme.dark ? 'dark' : 'light'],
+        theme
+      );
+    }
+
     const visual = (toggle ? (
-      <StyledCheckBoxToggle focus={focus} theme={theme}>
+      <StyledCheckBoxToggle focus={focus} theme={theme} checked={checked}>
         <StyledCheckBoxKnob theme={theme} />
       </StyledCheckBoxToggle>
     ) : (
-      <StyledCheckBoxBox focus={focus} theme={theme}>
+      <StyledCheckBoxBox
+        as={Box}
+        align='center'
+        justify='center'
+        width={theme.checkBox.size}
+        height={theme.checkBox.size}
+        border={{
+          size: theme.checkBox.border.width,
+          color: evalStyle(borderColor, theme),
+        }}
+        round={theme.checkBox.check.radius}
+        focus={focus}
+        theme={theme}
+        checked={checked}
+      >
         {
-          Icon ? <Icon /> : (
-            <svg viewBox='0 0 24 24' preserveAspectRatio='xMidYMid meet'>
+          Icon ? (
+            <Icon as={StyledCheckBoxIcon} theme={theme} checked={checked} />
```

Comment:
Why is `checked` being passed to the Icon?

Reply:
This is to be used by the `extend`. you may want to apply a different styled if this is checked or not. but I agree in the case of the icon, it will always be checked. I'm going to remove it.